man/systemd.unit.xml

```diff
@@ -1658,6 +1658,30 @@
           </listitem>
         </varlistentry>
 
+        <varlistentry>
+          <term><varname>ConditionMemoryPressure=</varname></term>
+          <term><varname>ConditionCPUPressure=</varname></term>
+
+          <listitem><para>Verify that the overall system (memory or CPU) pressure is below the given threshold.
+          This setting takes a colon-separated tuple as argument. The first value is the slice unit under which
+          the pressure will be checked, and it is optional: if not specified, the overall system pressure will be
+          measured, instead of a particular cgroup's.</para>
+
+          <para>The second value is the threshold. It can be specified as a simple percentage value, suffixed with
+          <literal>%</literal>, in which case the pressure will be measured as an average over the last five
+          minutes before the attempt to start the unit is performed, and it will be checked if it is strictly
+          below the specified threshold.</para>
```

Comment:
Hmm, do we really want to allow the slice specification? and why just slice? why not any unit?

If we allow referring other units we typically would add an implicit dep here, too. But i am not convinced that'd be right for a condition (as it's about silently skipping stuff, i.e. reducing stuff that is run, and pulling in new deps because of that sounds wrong).

I wonder: what is the precise usecase for allowing to specify arbitrary slices here? i.e. why would i want to start a service if some unrelated slice is under pressure or not?

I'd just remove this part I think: either always use the system as a whole or always look at the slice the unit itself is in. Or maybe add a "modifier" to choose between exactly these two slices. But allow arbitrary slices? I don#t see the usecase.

Reply:
Pulling a dependency definitely sounds wrong, so I avoided it intentionally.

This was asked here: https://github.com/systemd/systemd/pull/21437#pullrequestreview-810953380
I don't have a use case for this specific knob at the moment, but I think I can see a compelling one: consider the new app.slice/session.slice/background.slice we added for the user session. You could have a timer unit in the background.slice, that also checks the pressure in the app.slice and skips if it's above a threshold. That way you don't just reserve resources, but also ensure that the user is really doing nothing before starting to spin a core (and thus a fan) and so on.

Reply:
but why wouldn't be the root slice for that usecase OK too?

Reply:
That would be the sum of the children, no? Ie, if the root is under pressure, it might be because background.slice is, but in this case that's fine, it's the other you care about

Reply:
Yeah, it definitely sounds useful to be able to select an arbitrary slice (or even other unit) here. E.g. on server, start a task if the main payload is not under pressure.